extensions/sentry/src/sentry.ts

```diff
@@ -0,0 +1,57 @@
+import { getPreferenceValues } from "@raycast/api";
+import { useFetch } from "@raycast/utils";
+import fetch from "node-fetch";
+import { URLSearchParams } from "url";
+import { Event, Issue, Project, User } from "./types";
+
+const { token } = getPreferenceValues();
+const headers = { Authorization: `Bearer ${token}`, "Content-Type": "application/json" };
+
+export function useProjects() {
+  return useFetch<Project[]>("https://sentry.io/api/0/projects/", { headers });
+}
+
+export function useIssues(project?: Project) {
+  return useFetch<Issue[]>(`https://sentry.io/api/0/projects/${project?.organization?.slug}/${project?.slug}/issues/`, {
+    headers,
+    execute: !!project,
+  });
+}
+
+export function useIssueDetails(issue: Issue) {
+  return useFetch<Issue>(`https://sentry.io/api/0/issues/${issue.id}/`, { headers });
+}
+
+export function useLatestEvent(issue: Issue) {
+  return useFetch<Event>(`https://sentry.io/api/0/issues/${issue.id}/events/latest/`, {
+    headers,
+  });
+}
+
+export function useUsers(organizationSlug: string, projectId?: string) {
+  const searchParams = new URLSearchParams();
+  if (projectId) {
+    searchParams.append("project", projectId);
+  }
+
+  return useFetch<User[]>(`https://sentry.io/api/0/organizations/${organizationSlug}/users/?` + searchParams, {
```

Comment:
i -think- this introduces a new scope requirement. My old sentry API key, which has access to the scopes mentioned in the README (`event:read` and `project:read`) doesn't seem to work, or at least not fully:
<img width="645" alt="image" src="https://user-images.githubusercontent.com/1155589/186885330-9f6cd09d-162a-4210-a8b4-ab1072037699.png">

I generated an API key that has access to `org:read`, too, and that one seems to work as it should.

Reply:
I explained the new scopes in the README and also added a proper view for the unauthorized state. I hope this is enough for a "smooth" transition.